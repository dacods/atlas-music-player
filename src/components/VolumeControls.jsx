import volume from "../assets/Volume Icon.png"

export default function VolumeControls() {
  return (
    <div className="flex items-center w-full max-w-[400px] gap-3">
      <img src={volume} alt="Volume" className="h-5 w-5" />

      <input
        type="range"
        defaultValue="30"
        min="0"
        max="100"
        className="flex-1 h-2 bg-charcoal-gray dark:bg-deep-charcoal accent-beige dark:accent-dark-beige"
      />
    </div>
  )
}
