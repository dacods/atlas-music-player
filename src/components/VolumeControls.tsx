import volume from "../assets/Volume Icon.png"
import { useState, useEffect } from 'react';
interface VolumeControlsProps {
  setVolume: (volume: number) => void;
}


export default function VolumeControls({ setVolume }: VolumeControlsProps) {
  const [value, setValue] = useState(75);

  useEffect(() => {
    setVolume(value / 100);
  }, [value, setVolume]);

  return (
    <div className="flex items-center w-full max-w-[400px] gap-3">
      <img src={volume} alt="Volume" className="h-5 w-5" />

      <input
        type="range"
        defaultValue={value}
        onChange={(e) => setValue(Number(e.target.value))}
        min="0"
        max="100"
        className="flex-1 h-2 bg-charcoal-gray dark:bg-deep-charcoal accent-beige dark:accent-dark-beige"
      />
    </div>
  )
}
