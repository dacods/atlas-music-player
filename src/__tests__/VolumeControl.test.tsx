import { test, expect } from 'vitest';
import { render } from '@testing-library/react';
import VolumeControls from '../components/VolumeControls';

test('renders VolumeControls', () => {
    const { container } = render(<VolumeControls
    setVolume={() => {}}
    ></VolumeControls>);
    expect(container).toMatchSnapshot();
})