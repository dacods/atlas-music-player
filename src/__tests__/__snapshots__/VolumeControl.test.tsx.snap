// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders VolumeControls 1`] = `
<div>
  <div
    class="flex items-center w-full max-w-[400px] gap-3"
  >
    <img
      alt="Volume"
      class="h-5 w-5"
      src="/src/assets/Volume%20Icon.png"
    />
    <input
      class="flex-1 h-2 bg-charcoal-gray dark:bg-deep-charcoal accent-beige dark:accent-dark-beige"
      max="100"
      min="0"
      type="range"
      value="75"
    />
  </div>
</div>
`;
